opsdroid/configuration/schema.yaml

```diff
@@ -68,18 +68,18 @@ github:
 ---
 matrix:
   name: regex('^(?i)matrix$')
-  mxid: regex('^\@[a-zA-Z]*:matrix.org$')
+  mxid: str()
   password: str()
-  room: str()
+  room: str(required=False)
   rooms: (include('room1'))
-  homeserver: regex('^https://matrix.org$')
+  homeserver: str()
   nick: str(required=False)
   room_specific_nicks: bool(required=False)
   send_m_notice: bool(required=False)
 ---
 room1:
-  'main': regex('^#matrix:matrix.org')
-  'other': regex('^#riot:matrix.org$')
+  'main': str()
+  'other': str()
```

Comment:
Does this forbid other keys? if no, we probably don't need this, and if yes it wont work properly anyway :grin:

Reply:
No it doesn't. I checked it. Maybe the 'other' could be removed. But I would leave the 'main' as this forces you to have at least one room that is named 'main' which is needed by the connector if I'm right.

Reply:
Yeah I think that's the best option, keep the main since it's needed and just remove the other since they can be called whatever you want I believe

Reply:
yeah, the main should stay, I was referring to just the other line :smile:

Reply:
I'm still a bit unhappy with the solution. We have a field `room` which I tagged as not required (which is not super correct, it's required when `rooms` is not used but it's optional otherwise) and the 'main' entry of `rooms` is a required field.
Would it be ok to delete the `room` field? Then rooms can have 1 to n lines but it needs a line with the name 'main'.

Reply:
Do you mean delete the room field from the validation? I'm assuming that if you mark main as required it will always fail even if room is not required so I'd say that it's acceptable to remove the rooms field from the validation schema.

Also, I think the matrix connector already throws an error(or logs it) if there is no main room available.

Reply:
I am happy to drop room and just go with rooms, assuming @SolarDrew has no objections.